modules/web/src/app/wizard/step/cluster/component.ts

```diff
@@ -704,7 +706,9 @@ export class ClusterStepComponent extends StepBase implements OnInit, ControlVal
           [Controls.AuditPolicyPreset]:
             clusterSpec?.auditLogging?.policyPreset ?? this.controlValue(Controls.AuditPolicyPreset),
           [Controls.AuditWebhookBackend]: !!clusterSpec?.auditLogging?.webhookBackend,
-          [Controls.UserSSHKeyAgent]: clusterSpec?.enableUserSSHKeyAgent ?? this.controlValue(Controls.UserSSHKeyAgent),
+          [Controls.UserSSHKeyAgent]:
+            (this.isUserSshKeyEnabled && clusterSpec?.enableUserSSHKeyAgent) ??
+            this.controlValue(Controls.UserSSHKeyAgent),
```

Comment:
`??` would never be satisfied because `this.isUserSshKeyEnabled && clusterSpec?.enableUserSSHKeyAgent` would always be either `true` or `false`.
```suggestion
          [Controls.UserSSHKeyAgent]:
            (this.isUserSshKeyEnabled && clusterSpec?.enableUserSSHKeyAgent) ? true :
            this.controlValue(Controls.UserSSHKeyAgent),
```